src/runner/browser-job.js

```diff
@@ -99,7 +102,7 @@ export default class BrowserJob extends EventEmitter {
 
     // API
     get hasQueuedTestRuns () {
-        return !!this.testRunQueue.length;
+        return !!this.testRunQueue.length || this.opts.quarantineMode && !this.done;
```

Comment:
> With this approach we will always have one job left in the queue

No it won't be so, because next call of `job.hasQueuedTestRuns` will return `false` after calling `job.popNextTestRunUrl`, so `job` will be removed on next call of `browserConnection._popNextTestRunUrl()`. Looks like solution to me.